### PR TITLE
allow header and set header in client

### PR DESCRIPTION
**Pull request by hunjixin**
need send Authorization header to remote exec cmds. 
allow header and support set header value in client

### CODE REVIEW

**Comment by hunjixin** [comment]
@Stebalien

**Comment by hunjixin** [comment]
anyone can review this pr?

File: http/config.go
Comment (Stebalien) [inline http/config.go:91]: This should either be called `AddAllowedHeaders` or it should set (not append) the allowed headers.

File: http/client.go
Comment (Stebalien) [inline http/client.go:44]: Incorrect docs.

**Review by Stebalien — CHANGES_REQUESTED** [review]
The change seems reasonable, but please add a test in http/client_test.go (see the first test as a good example).

**Comment by BigLep** [comment]
@hunjixin: are you able to add the test so we can get this merged?

**Comment by hunjixin** [comment]
@BigLep  have tested

**Comment by BigLep** [comment]
Thanks.  We're blocked on finding maintainer review time.  It's in the queue though.

File: http/client.go
Comment (Stebalien) [inline http/client.go:45]: nit

```suggestion
// ClientWithHeader adds an HTTP header to the client.
func ClientWithHeader(key, value string) ClientOpt {
```

File: http/client_test.go
Comment (Stebalien) [inline http/client_test.go:129]:
```suggestion
		c := NewClient(tc.host, ClientWithHeader(tc.header, tc.value)).(*client)
```

**Review by Stebalien — APPROVED** [review]

**Comment by Stebalien** [comment]
Merged in https://github.com/ipfs/go-ipfs-cmds/pull/226 (addressing my review). Thanks!

**Pull request merged**
